README.md

```diff
@@ -283,6 +283,43 @@ Promise.all([builder.trace('app/first.js'), builder.trace('app/second.js')])
 });
 ```
 
+#### Example - Automatic Re-bundle Using a Watcher
+
+To watch for changes on the entire list of dependency files for a given entry point, and then
+automatically re-bundle when the files change, we can use the `trace()` method. This example uses
+`gulp` to watch, but you can use this pattern for any other watcher you like.
+
+```javascript
+var gulp = require('gulp');
+var path = require('path');
+
+var BASE_PATH = __dirname; // Replace with your SystemJS loader base path.
+var watcher;
+
+function watch(changed) {
+  if (changed) {
+    // The '*' is for matching SystemJS plugin syntax.
+    builder.invalidate(path.relative(BASE_PATH, changed.path) + '*');
+  }
+
+  if (watcher) {
+    watcher.end();
+  }
+
+  // List of dependencies may have changed, re-generate them with trace().
+  builder.trace('path/to/entryPoint.js').then(function (tree) {
+    var filesToWatch = Object.keys(tree).map(function (filePath) {
+      // Remove the SystemJS plugin syntax from the file path.
+      return path.join(BASE_PATH, filePath.split('!')[0]);
+    });
+
+    builder.bundle(tree);
+    watcher = gulp.watch(filesToWatch, watch);
+  });
+}
+watch();
```

Comment:
Thanks for posting. Unfortunately this feature is not yet stable so not planned to be documented until then.